packages/agoric-cli/src/commands/psm.js

```diff
@@ -0,0 +1,148 @@
+
+// Adapted from https://gist.github.com/dckc/8b5b2f16395cb4d7f2ff340e0bc6b610#file-psm-tool
+
+/**
+ * @template K, V
+ * @typedef {[key: K, val: V]} Entry<K,V>
+ */
+
+const last = xs => xs[xs.length - 1];
+
+/**
+ *
+ * @param {import('anylogger').Logger} logger
+ */
+export const makePsmCommand = async logger => {
+  const psm = new commander.Command('psm').description('PSM commands').usage(`
+
+  # prepare an offer to send
+  agops psm swap --wantMinted 6 --feePct 0.01 --offerId 123 > swap-123.json
+  
+  # sign and send
+  agd --node=https://xnet.rpc.agoric.net/ --chain-id=agoricxnet-13   --from=LEDGER_KEY_NAME --sign-mode=amino-json tx swingset wallet-action --allow-spend "$(cat swap-123.json)"
+
+  # watch for results
+  agoric wallet watch --from <wallet address or name>
+  `);
+
+  const { vstorage, fromBoard, agoricNames } = await makeRpcUtils({ fetch });
+
+  const getContractState = async () => {
+    const govContent = await vstorage.keys('published.psm.IST.AUSD.governance');
+    const { current: governance } = last(
+      storageNode.unserialize(govContent, fromBoard),
+    );
+    const { 'psm.IST.AUSD': instance } = agoricNames.instance;
+
+    return { instance, governance };
+  };
+
+  const showPublishedChildren = async () => {
+    // const status = await getJSON(`${RPC_BASE}/status?`);
+    // console.log({ status });
+    const raw = await getJSON(vstorage.url());
+    const top = vstorage.decode(raw);
+    console.error(
+      JSON.stringify(['vstorage published.*', JSON.parse(top).children]),
+    );
+  };
+
+  const showContractId = async showFees => {
+    const { instance, governance } = await getContractState();
+    showFees && console.error('psm', instance, Object.keys(governance));
+    showFees &&
+      console.error(
+        'WantMintedFee',
+        asPercent(governance.WantMintedFee.value),
+        '%',
+        'GiveMintedFee',
+        asPercent(governance.GiveMintedFee.value),
+        '%',
+      );
+    console.info(instance.boardId);
+  };
+  const marshaller = miniMarshal();
+
+  const lookupInstance = name => {
+    const instance = agoricNames.instance[name];
+    if (!instance) {
+      logger.debug('known instances:', agoricNames.instance);
+      throw new Error(`Unknown instance ${name}`);
+    }
+    return instance;
+  };
+
+  /** @param {import('../lib/psm.js').BridgeAction} bridgeAction */
+  const outputAction = bridgeAction => {
+    const capData = marshaller.serialize(bridgeAction);
+    process.stdout.write(JSON.stringify(capData));
+  };
+
+  psm
+    .command('swap')
+    .description('prepare an offer to swap')
+    .option(
+      '--wantMinted [DOLLARS]',
+      'amount of anchor tokens to give',
+      parseInt,
+    )
+    .option(
+      '--wantAnchor [DOLLARS]',
+      'amount of minted tokens to give',
+      parseInt,
+    )
+    .option('--feePct [%]', 'Gas fee percentage', parseInt)
+    .option('--offerId [number]', 'Offer id', parseInt, Date.now())
```

Comment:
This seems to work better:

```suggestion
    .option('--offerId [number]', 'Offer id', s => parseInt(s, 10), Date.now())
```